### PR TITLE
refactor(eval/funcs.c): resolve all clint errors

### PR DESCRIPTION
**Pull request by lewis6991**
... and tidy up local variable declarations

### CODE REVIEW

**Comment by clason** [comment]
Don't forget to add the the refactor commits to `.git-blame-ignore-revs` (probably best as a batch when done).

File: src/nvim/option_defs.h [hunk 1]
```diff
@@ -515,7 +515,7 @@ EXTERN char_u *p_lcs;         // 'listchars'
 
 EXTERN int p_lz;                // 'lazyredraw'
 EXTERN int p_lpl;               // 'loadplugins'
-EXTERN int p_magic;             // 'magic'
+EXTERN bool p_magic;            // 'magic'
```
Comment (zeertzjq) [inline src/nvim/option_defs.h:518]: No

Comment (zeertzjq) [inline src/nvim/option_defs.h:518]: No

File: src/nvim/eval/funcs.c [hunk 2]
```diff
@@ -2071,6 +2044,8 @@ static void f_expand(typval_T *argvars, typval_T *rettv, FunPtr fptr)
   const char *s = tv_get_string(&argvars[0]);
   if (*s == '%' || *s == '#' || *s == '<') {
     emsg_off++;
+    size_t len;
+    char_u *result;
     result = eval_vars((char_u *)s, (char_u *)s, &len, NULL, &errormsg, NULL);
```
Comment (zeertzjq) [inline src/nvim/eval/funcs.c:2049]:
```suggestion
    char_u *result = eval_vars((char_u *)s, (char_u *)s, &len, NULL, &errormsg, NULL);
```

File: src/nvim/eval/funcs.c [hunk 3]
```diff
@@ -2523,14 +2489,14 @@ static void f_foldtext(typval_T *argvars, typval_T *rettv, FunPtr fptr)
...
-    sprintf((char *)r, txt, dashes, count);
-    len = (int)STRLEN(r);
+    int count = foldend - foldstart + 1;
+    char *txt = NGETTEXT("+-%s%3ld line: ", "+-%s%3ld lines: ", count);
+    size_t len = STRLEN(txt)
+                 + STRLEN(dashes)  // for %s
+                 + 20              // for %3ld
+                 + STRLEN(s);      // concatenated
+    char_u *r = xmalloc(len);
+    snprintf((char *)r, len, txt, dashes, count);
```
Comment (zeertzjq) [inline src/nvim/eval/funcs.c:2499]:
```suggestion
    snprintf((char *)r, len, txt, dashes, count);
    len = (int)STRLEN(r);
```

File: src/nvim/eval/funcs.c [hunk 4]
```diff
@@ -8741,6 +8657,7 @@ static void f_spellsuggest(typval_T *argvars, typval_T *rettv, FunPtr fptr)
     maxcount = 25;
   }
 
+  garray_T ga = GA_EMPTY_INIT_VALUE;
```
Comment (zeertzjq) [inline src/nvim/eval/funcs.c:8660]: Better keep this at beginning of function to avoid using uninitialized memory.

File: src/nvim/eval/funcs.c [hunk 5]
```diff
@@ -2523,13 +2488,14 @@ static void f_foldtext(typval_T *argvars, typval_T *rettv, FunPtr fptr)
...
-    sprintf((char *)r, txt, dashes, count);
+    int count = foldend - foldstart + 1;
+    char *txt = NGETTEXT("+-%s%3ld line: ", "+-%s%3ld lines: ", count);
+    size_t len = STRLEN(txt)
+                 + STRLEN(dashes)  // for %s
+                 + 20              // for %3ld
+                 + STRLEN(s);      // concatenated
+    char_u *r = xmalloc(len);
+    snprintf((char *)r, len, txt, dashes, count);
     len = (int)STRLEN(r);
```
Comment (zeertzjq) [inline src/nvim/eval/funcs.c:2499]: forgot the type change
```suggestion
    len = STRLEN(r);
```

**Review by dundargoc — APPROVED** [review]

File: src/nvim/eval/funcs.c [hunk 6]
```diff
@@ -3014,7 +2976,7 @@ static void getpos_both(typval_T *argvars, typval_T *rettv, bool getcurpos, bool
       fp = &curwin->w_cursor;
     }
     if (fp != NULL && charcol) {
-      pos = *fp;
+      pos_T pos = *fp;
```
Comment (zeertzjq) [inline src/nvim/eval/funcs.c:2979]: Move this back. The scope of `pos` must not be narrower than `fp`.

Comment (zeertzjq) [inline src/nvim/eval/funcs.c]: Move this declaration back. The scope of `pos` must not be narrower than `fp`.

**Review by zeertzjq — CHANGES_REQUESTED** [review]

File: src/nvim/eval/funcs.c [hunk 7]
```diff
@@ -8795,6 +8711,7 @@ static void f_split(typval_T *argvars, typval_T *rettv, FunPtr fptr)
   };
   if (regmatch.regprog != NULL) {
     while (*str != NUL || keepempty) {
+      int match;
```
Comment (zeertzjq) [inline src/nvim/eval/funcs.c:8714]:
```suggestion
      bool match;
```

**Review by zeertzjq — APPROVED** [review]

**Pull request merged**